pkg/server/operands_handler.go

```diff
@@ -0,0 +1,102 @@
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+
+	"github.com/openshift/console/pkg/serverutils"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetConfig() (*rest.Config, error) {
+	// If an env variable is specified with the config location, use that
+	if len(os.Getenv("KUBECONFIG")) > 0 {
+		return clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	}
+	// If no explicit location, try the in-cluster config
+	if c, err := rest.InClusterConfig(); err == nil {
+		return c, nil
+	}
+	// If no in-cluster config, try the default location in the user's home directory
+	if usr, err := user.Current(); err == nil {
+		if c, err := clientcmd.BuildConfigFromFlags(
+			"", filepath.Join(usr.HomeDir, ".kube", "config")); err == nil {
+			return c, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not locate a kubeconfig")
+}
+
+func operandsListHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+	query := r.URL.Query()
+	operatorName := query.Get("name")
+	operatorNamespace := query.Get("namespace")
+	if len(operatorName) < 1 || len(operatorNamespace) < 1 {
+		errMsg := fmt.Sprintf("Failed to get operator name or namespace from the request URL: %q", r.URL.String())
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	scheme, err := action.NewScheme()
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to get new scheme for operator client: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	config, err := GetConfig()
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to get new config for operator client: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	client, err := client.New(config, client.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to get new client for listing operands: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	cfg := &action.Configuration{
+		Client:    client,
+		Scheme:    scheme,
+		Namespace: operatorNamespace,
+	}
+	operatorListOperands := action.NewOperatorListOperands(cfg)
+	operandsList, err := operatorListOperands.Run(context.TODO(), operatorName)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to list operands: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: errMsg})
```

Comment:
Bad request doesn't seem accurate for all cases here. If we need a general error and don't have a better status code, maybe 502 Bad Gateway?